memcached.c

```diff
@@ -5552,6 +5622,41 @@ static void process_misbehave_command(conn *c) {
         out_string(c, "OK");
     }
 }
+
+static void process_debugtime_command(conn *c, token_t *tokens, const size_t ntokens) {
+    struct timeval tv;
+
+    if (strcmp(tokens[1].value, "p") == 0) {
+        gettimeofday(&tv, NULL);
+        pthread_mutex_lock(&debugtime_lock);
+        if (!is_paused) {
+            paused_time = (rel_time_t)tv.tv_sec;
+            is_paused = true;
+        }
+        pthread_mutex_unlock(&debugtime_lock);
+    } else if (strcmp(tokens[1].value, "r") == 0) {
```

Comment:
see my other comments about fixing this: here you only apply the change on resume, but the workflow should be:

```
pause clock
run some tests
push time forward 3s
run more tests
push time forward another 2s
run even more tests
resume clock
```
which I don't believe works here since you only update the clock on resume, and I want it to stay paused while the tests go. It's good if the clock can be adjusted without having to pause first too, for hand debugging or other things.

I don't actually have a perfect idea offhand for how to do this cleanly. Can think about it more if you're stuck and I'm more caught up. probably just more logic to be done if `is_paused` is seen in the clock event.